cmd/kubeadm/app/apis/kubeadm/validation/validation.go

```diff
@@ -43,94 +47,153 @@ var cloudproviders = []string{
 	"vsphere",
 }
 
-func ValidateMasterConfiguration(c *kubeadm.MasterConfiguration) field.ErrorList {
-	allErrs := field.ErrorList{}
-	allErrs = append(allErrs, ValidateDiscovery(&c.Discovery, field.NewPath("discovery"))...)
-	allErrs = append(allErrs, ValidateServiceSubnet(c.Networking.ServiceSubnet, field.NewPath("service subnet"))...)
-	allErrs = append(allErrs, ValidateCloudProvider(c.CloudProvider, field.NewPath("cloudprovider"))...)
-	allErrs = append(allErrs, ValidateAuthorizationMode(c.AuthorizationMode, field.NewPath("authorization-mode"))...)
-	return allErrs
+type Error struct {
+	Msg string
+}
+
+func (e *Error) Error() string {
+	return fmt.Sprintf("[validation] Some errors occurred:\n%s", e.Msg)
+}
+
+func ValidateMasterConfiguration(c *kubeadm.MasterConfiguration) error {
+	allErrs := []error{}
+	allErrs = append(allErrs, ValidateServiceSubnet(c.Networking.ServiceSubnet)...)
+	allErrs = append(allErrs, ValidateCloudProvider(c.CloudProvider)...)
+	allErrs = append(allErrs, ValidateAuthorizationMode(c.AuthorizationMode)...)
+	if len(allErrs) > 0 {
+		var errs bytes.Buffer
+		for _, i := range allErrs {
+			errs.WriteString("\t" + i.Error() + "\n")
+		}
+		return &Error{Msg: errs.String()}
+	}
+	return nil
 }
 
-func ValidateNodeConfiguration(c *kubeadm.NodeConfiguration) field.ErrorList {
-	allErrs := field.ErrorList{}
-	allErrs = append(allErrs, ValidateDiscovery(&c.Discovery, field.NewPath("discovery"))...)
+func ValidateNodeConfiguration(c *kubeadm.NodeConfiguration) error {
+	allErrs := []error{}
+	allErrs = append(allErrs, ValidateDiscovery(c)...)
 
 	if !path.IsAbs(c.CACertPath) || !strings.HasSuffix(c.CACertPath, ".crt") {
-		allErrs = append(allErrs, field.Invalid(field.NewPath("ca-cert-path"), nil, "the ca certificate path must be an absolute path"))
```

Comment:
passing nil here was likely the cause of the panic... not sure I'd throw out the whole aggregate error object and re-invent an aggregate error